django/db/models/sql/query.py

```diff
@@ -1296,8 +1296,7 @@ def build_filter(self, filter_expr, branch_negated=False, current_negated=False,
                 #   <=>
                 # NOT (col IS NOT NULL AND col = someval).
                 lookup_class = targets[0].get_lookup('isnull')
-                col = _get_col(targets[0], join_info.targets[0], alias, simple_col)
-                clause.add(lookup_class(col, False), AND)
+                clause.add(lookup_class(condition.lhs, False), AND)
```

Comment:
As far as I'm concerned all builtin lookups preserve `NULL` values, so I maybe we should add some flag to mark transforms that can change `NULL`'s and use `condition.lhs` only in such cases :thinking: .

Reply:
yeah having an `Expression.nullable` flag that is also present on `Col` instance based on `Field.null` they resolve would be useful for a few other things I've worked on in the past.

Reply:
@charettes Agreed, what do you think also about flag about preserving `NULL`'s? Running functions can be expensive (maybe not on `NULL`'s :thinking:) and for all builtin lookups this is unnecessary because `FUNCTION(column) IS NULL` is equivalent to `column IS NULL`.

Reply:
I pushed a commit. is it like this?